source/Public/New-GuestConfigurationPolicy.ps1

```diff
@@ -318,14 +347,28 @@ function New-GuestConfigurationPolicy
     $tempPath = Reset-GCWorkerTempDirectory
     $packagePath = Join-Path -Path $tempPath -ChildPath 'extracted'
 
-    if (-not ([string]::IsNullOrWhiteSpace($ManagedIdentityResourceId) -or [string]::IsNullOrWhiteSpace($LocalContentPath)))
+    # This means the customer wants to use either identity to download the package.
+    if (-not [string]::IsNullOrWhiteSpace($LocalContentPath))
     {
-        if (-not $ExcludeArcMachines)
+        # This means the customer wants to use the User-Defined id.
+        if (-not [string]::IsNullOrWhiteSpace($ManagedIdentityResourceId))
         {
-            throw "The ManagedIdentityResourceId and LocalContentPath parameters are defined but the -ExcludeArcMachines parameter is not. Managed identities cannot be used with Azure Arc machines. Please provide the -ExcludeArcMachines parameter to exclude Azure Arc machines and use a managed identity with this policy."
-        }
+            if (-not $ExcludeArcMachines)
+            {
+                throw "The ManagedIdentityResourceId and LocalContentPath parameters are defined but the -ExcludeArcMachines parameter is not. User assigned managed identities cannot be used with Azure Arc machines. Please provide the -ExcludeArcMachines parameter to exclude Azure Arc machines and use a managed identity with this policy."
+            }
 
-        $packageFileDownloadPath = $LocalContentPath
+            $packageFileDownloadPath = $LocalContentPath
+        }
+        # This means the customer wants to use the System-Assigned id.
+        elseif ($UseSystemAssignedIdentity)
+        {
+            $packageFileDownloadPath = $LocalContentPath
+        }
+        else
+        {
+            throw "The LocalContentPath is defined but either of the identity is not given. Please provide ManagedIdentityResourceId along with ExcludeArcMachine or use flag UseSystemAssignedIdentity."
```

Comment:
This error message doesn't make sense. I think you wanted it to say neither identity parameter is given?
Also I feel like this is redundant with the top checks?